tests/constraints/test_constraints_continuous.py

```diff
@@ -68,6 +68,81 @@ def test_inequality3(campaign, n_iterations, batch_size):
     assert (1.0 * res["Conti_finite1"] + 3.0 * res["Conti_finite2"]).le(0.301).all()
 
 
+@pytest.mark.parametrize("parameter_names", [["Conti_finite1", "Conti_finite2"]])
+@pytest.mark.parametrize("constraint_names", [["InterConstraint_1"]])
+@pytest.mark.parametrize("batch_size", [5], ids=["b5"])
+def test_interpoint_equality_single_parameter(campaign, n_iterations, batch_size):
+    """Test single parameter inter-point equality constraint."""
+    run_iterations(campaign, n_iterations, batch_size, add_noise=False)
+    res = campaign.measurements
+    print(res)
+
+    for batch in range(n_iterations):
+        res_batch = res[res["BatchNr"] == batch + 1]
+        assert np.isclose(res_batch["Conti_finite1"].sum(), 0.3)
+
+
+@pytest.mark.parametrize("parameter_names", [["Conti_finite1", "Conti_finite2"]])
+@pytest.mark.parametrize("constraint_names", [["InterConstraint_2"]])
+@pytest.mark.parametrize("batch_size", [5], ids=["b5"])
+def test_interpoint_inequality_single_parameter(campaign, n_iterations, batch_size):
+    """Test single parameter inter-point inequality constraint."""
+    run_iterations(campaign, n_iterations, batch_size, add_noise=False)
+    res = campaign.measurements
+    print(res)
+
+    for batch in range(n_iterations):
+        res_batch = res[res["BatchNr"] == batch + 1]
+        assert 2 * res_batch["Conti_finite1"].sum() >= 0.299
+
+
+@pytest.mark.parametrize("parameter_names", [["Conti_finite1", "Conti_finite2"]])
+@pytest.mark.parametrize("constraint_names", [["InterConstraint_3"]])
+@pytest.mark.parametrize("batch_size", [5], ids=["b5"])
+def test_interpoint_equality_multiple_parameters(campaign, n_iterations, batch_size):
+    """Test inter-point equality constraint involving multiple parameters."""
+    run_iterations(campaign, n_iterations, batch_size, add_noise=False)
+    res = campaign.measurements
+    print(res)
+
+    for batch in range(n_iterations):
+        res_batch = res[res["BatchNr"] == batch + 1]
+        assert np.isclose(
+            res_batch["Conti_finite1"].sum() + res_batch["Conti_finite2"].sum(), 0.3
+        )
+
+
+@pytest.mark.parametrize("parameter_names", [["Conti_finite1", "Conti_finite2"]])
+@pytest.mark.parametrize("constraint_names", [["InterConstraint_4"]])
+@pytest.mark.parametrize("batch_size", [5], ids=["b5"])
+def test_interpoint_inequality_multiple_parameters(campaign, n_iterations, batch_size):
```

Comment:
Well, I followed the structure that we already had, where there is a dedicated test for each individual constraint. I would thus stay with the current design since it is more in line with the current design and better readable in my opinion.

Reply:
not sure where thats consistent, dont they all test the same constaint just a different configuration? thats pretty much a use case for parameterization

Reply:
"Consistent" in the sense of "there is a dedicated test per constraint". Also, we do the very same thing for other constraints, e.g. here:
```python
@pytest.mark.parametrize("parameter_names", [["Conti_finite1", "Conti_finite2"]])
@pytest.mark.parametrize("constraint_names", [["ContiConstraint_3"]])
@pytest.mark.parametrize("batch_size", [5], ids=["b5"])
def test_inequality1(campaign, n_iterations, batch_size):
    """Test inequality constraint with equal weights."""
    run_iterations(campaign, n_iterations, batch_size, add_noise=False)
    res = campaign.measurements
    print(res)
    assert (1.0 * res["Conti_finite1"] + 1.0 * res["Conti_finite2"]).ge(0.299).all()
@pytest.mark.parametrize("parameter_names", [["Conti_finite1", "Conti_finite2"]])
@pytest.mark.parametrize("constraint_names", [["ContiConstraint_4"]])
@pytest.mark.parametrize("batch_size", [5], ids=["b5"])
def test_inequality2(campaign, n_iterations, batch_size):
    """Test inequality constraint with unequal weights."""
    run_iterations(campaign, n_iterations, batch_size, add_noise=False)
    res = campaign.measurements
    print(res)
    assert (1.0 * res["Conti_finite1"] + 3.0 * res["Conti_finite2"]).ge(0.299).all()
```
Also, do I understand it correctly that you would then prefer to have the code then in the style of 
```python

    for batch in range(n_iterations):
        res_batch = res[res["BatchNr"] == batch + 1]
        assert campaign.searchspace.constraints[0].coefficients[0] * res_batch["Conti_finite1"].sum() >= campaign.searchspace.constraints[0].rhs
```
or similar? In my opinion, that is significantly harder to read, so I am still not convinced that the design here should change.

Reply:
ok fine, my suggestion was coming form the persepctive of reducing code (whihc i think is possible here by a factor of 3 to 4 or so) but I guess it would reduce readability as you said